src/extension.ts

```diff
@@ -1067,6 +1065,34 @@ const getActivePythonFile = () => {
     return activeEditor ? activeEditor.document.fileName : "";
 };
 
+const updateCurrentFileFromEditor = async (
+    activeTextDocument: vscode.TextEditor | undefined,
+    currentPanel: vscode.WebviewPanel
+) => {
+    if (
+        activeTextDocument &&
+        activeTextDocument.document &&
+        activeTextDocument.document.languageId === "python"
+    ) {
+        setPathAndSendMessage(
+            currentPanel,
+            activeTextDocument.document.fileName
+        );
+        currentTextDocument = activeTextDocument.document;
+    } else if (currentFileAbsPath === "") {
+        setPathAndSendMessage(currentPanel, getActivePythonFile() || "");
+    }
+    if (
```

Comment:
If `currentFileAbsPath`, should we check for `currentTextDocument`? I'm not entirely familiar with the code flow at this point, so I might have missed something. It might be worth adding a comment about this too.

Reply:
No since currentTextDocument might be outdated at this point, we want to call getActivePythonFile to set a new currentTextDocument